src/proxy.cc

```diff
@@ -580,7 +580,7 @@ static ncclResult_t ncclProxyGetPostedOps(struct ncclComm* comm, int* added) {
       int profId;
       NCCLCHECK(ncclProfEventStart(comm->proxyState.profilerRecord, &profId, PROF_TYPE_SLEEP, 0));
       pthread_cond_wait(&pool->cond, &pool->mutex);
-      ncclProfEventTime(comm->proxyState.profilerRecord, profId, NCCL_PROF_TS_END);
+      NCCLCHECK(ncclProfEventTime(comm->proxyState.profilerRecord, profId, NCCL_PROF_TS_END));
```

Comment:
Is the NCCLCHECK change a bug fix which should be moved to the other commit?

Reply:
No - when I converted ncclProfEventTime, I also changed it to return a success error code.

Design justification-

Since the return comes after the timestamp is logged and this should only add one branch to check the result, I figured any change to results should be nonexistent to negligible, but this allows for more flexibility in the plugin if it needs it. It's also consistent with the other API functions which have more potential to fail (ncclProfEventTime seems to be solely for recording a single timestamp, so it's probably the most deterministic part of the new profiler API).

I did consider deliberately leaving ncclProfEventTime as a built-in function so that you would have ncclProfEventStart (customizable but plugin call overhead) and ncclProfEventTime (not customizable but no overhead), but I believe this would implicitly force ncclProfEventStart to fill out the built-in data structures whether it's using them or not in order to prevent ncclProfEventTime from crashing.

I think we can use the default NVIDIA data structures, but I think the flexibility would be more important to us than sub-microsecond accuracy. Since this is a part of the plugin API that will be consumed by multiple people and will have to be backwards compatible, I decided in favor of the option that gave us the most flexibility rather than leaving it alone until absolutely necessary (at which point it might be more difficult to request a change because it would require a new version or breaking things for other people).